examples/chub/client.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import "github.com/spf13/cobra"
+
+const (
+	// these are needed for every init
+	flagChainID = "chain-id"
+	flagNode    = "node"
+
+	// one of the following should be provided to verify the connection
+	flagGenesis = "genesis"
+	flagCommit  = "commit"
+	flagValHash = "validator-set"
+
+	flagSelect = "select"
+	flagTags   = "tag"
+	flagAny    = "any"
+
+	flagBind      = "bind"
+	flagCORS      = "cors"
+	flagTrustNode = "trust-node"
+
+	// this is for signing
+	flagName = "name"
+)
+
+var (
+	statusCmd = &cobra.Command{
+		Use:   "status",
+		Short: "Query remote node for status",
+		RunE:  todoNotImplemented,
+	}
+
+	getCmd = &cobra.Command{
+		Use:   "get",
+		Short: "Query ABCI state data",
+		Long: `Query ABCI state data.
+Subcommands should be defined for each particular object to query.`,
+	}
+
+	postCmd = &cobra.Command{
+		Use:   "post",
+		Short: "Create a new transaction and post it to the chain",
+		Long: `Create a new transaction and post it to the chain.
+Subcommands should be defined for each particular transaction type.`,
+	}
+)
+
+func init() {
+	getCmd.PersistentFlags().Bool(flagTrustNode, false, "Don't verify proofs for responses")
+
+	postCmd.PersistentFlags().String(flagName, "", "Name of private key with which to sign")
+	postCmd.PersistentFlags().String(flagPassword, "", "Password to use the named private key")
+}
+
+// ClientCommands returns a sub-tree of all basic client commands
+//
+// Call AddGetCommand and AddPostCommand to add custom txs and queries
+func ClientCommands() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "client",
+		Short: "Interact with the chain via a light-client",
+	}
+	cmd.AddCommand(
+		initClientCommand(),
```

Comment:
👍  on the design where you initialize the flags in the same location as defining the command, aesthetically I want to say that we should define all of the command in functions (such as the ones where flags are also being initialized) but It doesn't really matter... it was just confusing for a second seeing some commands in this list setup as functions and some without the `()`

Reply:
OMG yes so much this. Digging in tmlibs for flag settings is no fun at all